ghcide/src/Development/IDE/Core/Shake.hs

```diff
@@ -437,9 +438,9 @@ deleteValue
   -> k
   -> NormalizedFilePath
   -> IO ()
-deleteValue ShakeExtras{dirtyKeys, state} key file = do
-    void $ modifyVar' state $ HMap.delete (toKey key file)
-    atomicModifyIORef_ dirtyKeys $ HSet.insert (toKey key file)
+deleteValue ShakeExtras{dirtyKeys, state} key file = atomically $ do
```

Comment:
this one should maybe have got pushed to the call sites too? particularly paralleling `setValues`